### DISCUSSION TITLE
Using a MUI Icon as side label text

### DISCUSSION THREAD

**Discussion by ndtreviv** [f5/unovis#106] [Q&A] [answered]
I want to have a side label with a [Material UI Icon](https://mui.com/material-ui/material-icons/) in it.

This is what I'm trying:

```
          nodeSideLabels={useCallback((n) => n.expanded ? [] : [{ text: <AllOutIcon /> }], [])}
```
but when it renders the side label it doesn't render the icon, but instead: `[object Object]`. I can see why, too.

My question is: Is there a way to make this work?
If not, it would be a nice feature :)

**Comment by rokotyan** [comment, marked as answer]
Since Unovis's core is not native to React, you can't provide icons as React components. However, you can import Material UI icons as an [icon font](https://developers.google.com/fonts/docs/material_icons) and provide an the exact characters code to `text`:
```
          nodeSideLabels={useCallback((n) => n.expanded ? [] : [{ text: `&#xE87C;` }], [])}
```
Don't forget to set the `--vis-graph-icon-font-family` CSS variable to the name of your icon font, e.g.: 
```css
--vis-graph-icon-font-family : 'Material Icons'
```


P.S. You can browse through the icons and find their character codes [here](https://fonts.google.com/icons):
b1.png)
![Screen Shot 2023-01-19 at 6 45 23 AM](https://user-images.githubusercontent.com/755708/213473796-5c7da827-8504-48f7-955b-00f01e377d4a.png)

**Reply by ndtreviv** [reply]
Excellent - thanks!

**Reply by ndtreviv** [reply]
Actually, I just noticed that the one I'm using (https://mui.com/material-ui/material-icons/) renders them as SVGs rather than using a special font. I'm guessing that won't work...?

**Reply by rokotyan** [reply]
@ndtreviv Unfortunately not, we'll think about adding support for SVG icons in the future though.
